### ISSUE TITLE
Incorrect test: built-ins/TypedArray/prototype/set/typedarray-arg-set-values-same-buffer-other-type.js

### ISSUE THREAD

**Issue by mkaric** [tc39/test262#757]
This test initializes a Float64Array with elements [1, 2, 3, 4, 5, 6, 7, 8]. Then the same buffer is used to create a Float32Array of size 2 (covering only the first element of the Float64Array).
So the binary representation of the Float32Array is the following:
```
little endian
00 00 00 00 00 00 F0 3F
big endian
3F F0 00 00 00 00 00 00
```
Then the first element of the Float32Array is assigned the value 42, this results in:
```
little endian
00 00 28 42 00 00 F0 3F
big endian
42 28 00 00 00 00 00 00
```
The test relies on little endian architecture, on big endian it always fails as the initial bits are overwritten. Writing 42 to the second element of the Float32Array and adjusting expected values for big endian would make the test pass.

**Comment by leobalter** [comment]
cc @bakkot @syg Can you help confirming this one?

@mkaric I believe there's an error reading the test you mentioned, can you confirm this is still incorrect?

**Comment by mkaric** [comment]
Yes, it is still incorrect. I just tested again and it fails. It is the same version of the test as when I reported the issue.

**Comment by leobalter** [comment]
@mkaric you're right. Most of the TypedArray tests checking for bit precision are relying on little endian order while it is implementation dependent.

Would you help me specifying which env are you using to verify the big endian order? It would help me checking the results when fixing these.

---

For now, I will have to flag all the tests relying on the little endian order so they can be filtered by runners until we cover with the other tests.

**Comment by mkaric** [comment]
I have access to a big endian mips board. What I do is build v8 for mips, ssh to mips big endian board and run the test. But I just ran the test using QEMU (big endian mips user-mode emulation) and it produces the same failure, so you could use this as well. If you would like access to a mips board I can check if there is a way to help you with that.

**Comment by leobalter** [comment]
It would be great! It provides me something to verify my calculations for bit precision values on every test.

**Comment by mkaric** [comment]
This might give you some info on how to do this:
https://github.com/v8mips/v8mips/wiki/Building-v8-for-BE-with-QEMU

I will get back to you tomorrow with the best solution for the environment, perhaps I can make a script that will create the environment for you and run the test in matter (on linux).

**Comment by leobalter** [comment]
Thank you

**Comment by mkaric** [comment]
Hi and sorry for the delay. Here is a script that sets the environment and runs the test using big endian mips emulation. It might work as it is but you may need to make some adjustments. Some dependencies might still be missing. Give it a try and let me know what happens. 
[MakeEnvRunTestBE.sh.txt](https://github.com/tc39/test262/files/847444/MakeEnvRunTestBE.sh.txt)

**Comment by mkaric** [comment]
@leobalter just a remider in case you missed my message, please try the script and let me know how it works.

**Comment by leobalter** [comment]
@mkaric Thanks for the reminder! I'm just stuck in too many tasks right now I didn't have time to try it yet. It also reminds me I'll need to come with a plan to fix the tests relying on little endianness order only.

**Comment by suve** [comment]
I've recently hit this issue as well and wrote a patch to use different expected values for little-endian and big-endian. Do you have some requirements or expectations regarding a possible solution to this issue, e.g. proving that the values are correct?

**Comment by ptomato** [comment]
@suve Sorry for the delay in replying. I don't know what your patch looks like, feel free to submit it and get feedback! I think we'd want to test that the result matches either the expected big-endian or little-endian value, and that the results are consistent (i.e. the implementation produces only big-endian or only little-endian results)
